libraries/AP_Camera/AP_Camera_Servo.cpp

```diff
@@ -61,19 +61,31 @@ bool AP_Camera_Servo::trigger_pic()
 
 bool AP_Camera_Servo::set_zoom(ZoomType zoom_type, float zoom_value)
 {
-    if (zoom_type == ZoomType::RATE) {
-        zoom_current_rate = zoom_value;
-        return true;
+    switch (zoom_type) {
+        case ZoomType::RATE:
+            zoom_current_rate = zoom_value;
+            return true;
+        case ZoomType::PCT:
+            // expects to receive a value between 0 and 100
+            SRV_Channels::set_output_scaled(SRV_Channel::k_cam_zoom, constrain_float(zoom_value * 10, 0, 1000));
+            return true;
     }
     return false;
```

Comment:
```suggestion
        default:
            return false;
    }
```
Not sure what ArduPilot's convention is for function-ending `switch-case` blocks, but I think it's potentially clearer to have the final result handled in `default` instead of outside the block 🤷‍♂️

Reply:
I can't speak for the dev team, but I prefer not to have a `default` and to handle unexpected values outside of the `switch`. That way, when a new `enum` value is added, the compiler will warn that not all cases are handled.